File: ui/hub-mk3-ui/src/store/orm/program-meme.js
// Copyright (c) XJ Music Inc. (https://xj.io) All Rights Reserved.
import {attr, fk, Model} from "redux-orm";

class ProgramMeme extends Model {
  toString() {
    return `ProgramMeme: ${this.name}`;
  }

  // Declare any static or instance methods you need.
}

ProgramMeme.modelName = 'ProgramMeme';

// Declare your related fields.
ProgramMeme.fields = {
  id: attr(),
  name: attr(),
  programId: fk({
    to: 'Program',
    as: 'program',
    relatedName: 'memes',
  }),
};

export default ProgramMeme;

File: engine/src/content/ProgramMeme.cpp
// Copyright (c) XJ Music Inc. (https://xjmusic.com) All Rights Reserved.

#include "xjmusic/content/ProgramMeme.h"

using namespace XJ;

std::set<std::string> ProgramMeme::getNames(const std::set<const ProgramMeme *> &programMemes) {
  std::set<std::string> names;
  for (const auto &programMeme: programMemes) {
    names.insert(programMeme->name);
  }
  return names;
}
